src/groupByKey.ts

```diff
@@ -1,7 +1,19 @@
-type GroupsMap<T> = {
+type GroupsMap <T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const gropedByKey: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const currrentKey = String(item[key]);
+
+    if (!gropedByKey[currrentKey]) {
+      gropedByKey[currrentKey] = [item];
+    } else {
+      gropedByKey[currrentKey].push(item);
+    }
```

Comment:
you can invert your condition to just **add blank array**, and **then** push your item to get rid of `else`